app/models/application_record.rb

```diff
@@ -13,4 +13,12 @@ class ApplicationRecord < ActiveRecord::Base
   # FIXME: UI code - decorator support
   extend MiqDecorator::Klass
   include MiqDecorator::Instance
+
+  # API Support
+  virtual_column :slug, :type => :string
+
+  def slug
+    collection = Api::CollectionConfig.new.name_for_subklass(self.class)
```

Comment:
Suggest returning early if `id` is `nil` so you do not spend time in `name_for_subklass` for new records that do not have an ID yet.

```ruby
def slug
  return nil unless id
  collection = Api::CollectionConfig.new.name_for_subklass(self.class)
  "#{collection}/#{id}" if collection
end
```